core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -56,16 +58,22 @@ import shapeless.ops.hlist.Tupler
  *   Http.server.serve(foobar.toService)
  * }}}
  */
-trait Endpoint[A] { self =>
+abstract class Endpoint[A] extends Serializable { self =>
 
-  type ContentType <: String
+  /**
+   * The [[StateT]] instance underneath this endpoint.
+   */
+  def run: Endpoint.State[A]
```

Comment:
Yeah, I don't think `run` is terrible, but I don't personally find the convention quite as clear here. I don't feel strongly either way.